zerver/lib/markdown/fenced_code.py

```diff
@@ -364,6 +369,37 @@ def done(self) -> None:
         self.processor.pop()
 
 
+class KrokiHandler(BaseHandler):
+    def __init__(
+        self,
+        processor: Any,
+        output: MutableSequence[str],
+        fence: str,
+        diagram_type: str,
+    ) -> None:
+        self.processor = processor
+        self.output = output
+        self.fence = fence
+        self.diagram_type = diagram_type
+        self.lines: List[str] = []
+
+    def handle_line(self, line: str) -> None:
+        if line.rstrip() == self.fence:
+            self.done()
+        else:
+            self.lines.append(line)
```

Comment:
Does it makes sense to do a preparatory refactor PR to move functions like this into `BaseHandler` or perhaps a new `ZulipBaseHandler` class?  It feels like most of the code of these are duplicated.